jdaviz/core/template_mixin.py

```diff
@@ -440,7 +439,7 @@ def _update_subset(self, subset, attribute=None):
                 # NOTE: += will not trigger traitlet update
                 self.items = self.items + [self._subset_to_dict(subset)]  # noqa
         else:
-            if attribute in ('style'):
+            if attribute in ('style', 'type'):
```

Comment:
clever!  Can you please add a comment that `'type'` is passed manually and not from the `SubsetUpdateMessage.attribute` (I'm just worried we might accidentally remove it in the future or not remember why its necessary)?